File: C#2019/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication1
{

 
    public partial class Form1 : Form
    {
        string strApp;
        public Form1()
        {
            InitializeComponent();
        }

        private void BtnSourceImage_Click(object sender, EventArgs e)
        {

            OpenFileDialog1.Filter = "All Files (*.*)|*.*|JPEG (*.jpg)|*.jpg|PNG (*.png)|*.png|Bitmap (*.bmp)|*.bmp|TIF (*.tif)|*.tif";

            if(OpenFileDialog1.ShowDialog() == DialogResult.OK)
            {
                txtsourceimage.Text = OpenFileDialog1.FileName;

            }

        }

        private void RadioButton1_Click(object sender, EventArgs e)
        {
              txtmaxsizewidth.Text = "0";
              txtmaxsizeheight.Text = "0";
              BtnSourceImage.Enabled = true;
        }

        private void RadioButton2_Click(object sender, EventArgs e)
        {
            txtmaxsizewidth.Text = "50";
            txtmaxsizeheight.Text = "50";
            BtnSourceImage.Enabled = false;
        }

        private void RadioButton3_Click(object sender, EventArgs e)
        {
            txtmaxsizewidth.Text = "0";
            txtmaxsizeheight.Text = "0";
            BtnSourceImage.Enabled = true;
        }

        private void BtnOutputImage_Click(object sender, EventArgs e)
        {
             SaveFileDialog1.Filter = "PNG (*.png)|*.png";

            if(SaveFileDialog1.ShowDialog() == DialogResult.OK)
            {
                txtoutputimage.Text = SaveFileDialog1.FileName;
            }
        }

        private void chkoutputdetectedimage_Click(object sender, EventArgs e)
        {
            if(chkoutputdetectedimage.Checked)
            {
                txtoutputimage.Enabled = true;
                BtnOutputImage.Enabled = true;
            }
            else
            {
                txtoutputimage.Enabled = false;
                BtnOutputImage.Enabled = false;
            }
      
        }

        private void Form1_Load(object sender, EventArgs e)
        {
              int iFindIndex;
            strApp = Application.ExecutablePath;

           
            iFindIndex = strApp.IndexOf("Examples");

            if (iFindIndex != -1)
            {

                strApp = strApp.Substring(0, iFindIndex);

                strApp = strApp + "FacePhoto";

             

                cbomingeighors.Items.Add("1");
                cbomingeighors.Items.Add("2");
                cbomingeighors.Items.Add("3");
                cbomingeighors.Items.Add("4");
                cbomingeighors.Items.Add("5");
                cbomingeighors.SelectedIndex = 3;
            }

        }

        private void ControlCountFace()
        {
            if(RadioButton1.Checked)
              lbltotalface.Text = axFaceDetectionControl1.CountFaces(strApp + "\\demo1.png").ToString();

           if(RadioButton2.Checked)
             lbltotalface.Text =axFaceDetectionControl1.CountFaces(strApp + "\\demo2.png").ToString();

           if (RadioButton3.Checked)
               lbltotalface.Text = axFaceDetectionControl1.CountFaces(txtsourceimage.Text).ToString();

        }
        private void BlackList()
        {
          int ix;
          int iy;
          int iwidth;
           int iheight; 

             axFaceDetectionControl1.ClearBlackList();

          
            for (int i = 0; i < axFaceDetectionControl1.DetectedRectGetCount(); i++)
            {
                ix = axFaceDetectionControl1.DetectedRectGetX((short)i);
                iy = axFaceDetectionControl1.DetectedRectGetY((short)i);
                iwidth = axFaceDetectionControl1.DetectedRectGetWidth((short)i);
                iheight = axFaceDetectionControl1.DetectedRectGetHeight((short)i);
               
                if(i==0 || i==1)
                     axFaceDetectionControl1.Add2BlackList((short)ix, (short)iy, (short)iwidth, (short)iheight);
            }
           
          


        }

        private void Button2_Click(object sender, EventArgs e)
        {
            if(RadioButton3.Checked)
            {
                if(txtsourceimage.Text.Length==0)
                {
              
                    MessageBox.Show("Please input source image first");
                    return;

                }
            }

            if(chkoutputdetectedimage.Checked)
            {
                if(txtoutputimage.Text == "")
                {
                    MessageBox.Show("Please select the path of output detected image first");
                    return;
                }
            }


        axFaceDetectionControl1.SetFrontalFaceCascade(txtfrontalcascade.Text);
        axFaceDetectionControl1.SetProfileFaceCascade(txtprofilecascade.Text);


        axFaceDetectionControl1.EnableFrontalFace = chkfrontalface.Checked;
        axFaceDetectionControl1.EnableProfileFace = chkprofileface.Checked;
        axFaceDetectionControl1.EnableTiltedFrontalFace = chktiltedfrontalface.Checked;
        axFaceDetectionControl1.EnableTiltedProfileFace = chktiltedprofileface.Checked;
        axFaceDetectionControl1.EnableEqualizeHistogram = chkequalize.Checked;
     
          
       if( txtoutputimage.Text != "")
            axFaceDetectionControl1.SetSaveResults(txtoutputimage.Text);


        axFaceDetectionControl1.SetMinNeighbors(Int32.Parse(cbomingeighors.Text));
        axFaceDetectionControl1.SetScaleFactor(Double.Parse( txtscalefactor.Text));
        axFaceDetectionControl1.SetMinSize(Int32.Parse(txtminsizewidth.Text), Int32.Parse(txtminsizeheight.Text));

        axFaceDetectionControl1.SetMaxSize(Int32.Parse(txtmaxsizewidth.Text), Int32.Parse(txtmaxsizeheight.Text));


        axFaceDetectionControl1.ClearBlackList();
        ControlCountFace();
       


         System.IO.FileStream fs;
       //Specify a valid picture file path on your computer.

        if(txtoutputimage.Text != "")
        {
            if(System.IO.File.Exists( txtoutputimage.Text))
            {
                fs = new System.IO.FileStream(txtoutputimage.Text,
                System.IO.FileMode.Open, System.IO.FileAccess.Read);
                PictureBox1.Image = System.Drawing.Image.FromStream(fs);
                fs.Close();
            }
        }



        ListBox1.Items.Clear();
            int ix, iy,iwidth,iheight;


         
        for( int i = 0 ; i< axFaceDetectionControl1.DetectedRectGetCount() ; i++)
        {
            ix =axFaceDetectionControl1.DetectedRectGetX((short)i);
            iy = axFaceDetectionControl1.DetectedRectGetY((short)i);
            iwidth =axFaceDetectionControl1.DetectedRectGetWidth((short)i);
            iheight = axFaceDetectionControl1.DetectedRectGetHeight((short)i);

            ListBox1.Items.Add(ix.ToString() + " " + iy.ToString() + " " + iwidth.ToString() + " " + iheight.ToString());
        }

              btnblacklist.Enabled =true;
        }

        private void btnblacklist_Click(object sender, EventArgs e)
        {

            if (RadioButton3.Checked)
            {
                if (txtsourceimage.Text.Length == 0)
                {

                    MessageBox.Show("Please input source image first");
                    return;

                }
            }

            if (chkoutputdetectedimage.Checked)
            {
                if (txtoutputimage.Text == "")
                {
                    MessageBox.Show("Please select the path of output detected image first");
                    return;
                }
            }


            axFaceDetectionControl1.SetFrontalFaceCascade(txtfrontalcascade.Text);
            axFaceDetectionControl1.SetProfileFaceCascade(txtprofilecascade.Text);



            axFaceDetectionControl1.SetFrontalFaceCascade(txtfrontalcascade.Text);
            axFaceDetectionControl1.SetProfileFaceCascade(txtprofilecascade.Text);


            axFaceDetectionControl1.EnableFrontalFace = chkfrontalface.Checked;
            axFaceDetectionControl1.EnableProfileFace = chkprofileface.Checked;
            axFaceDetectionControl1.EnableTiltedFrontalFace = chktiltedfrontalface.Checked;
            axFaceDetectionControl1.EnableTiltedProfileFace = chktiltedprofileface.Checked;
            axFaceDetectionControl1.EnableEqualizeHistogram = chkequalize.Checked;
     
         

           
            if (txtoutputimage.Text != "")
                axFaceDetectionControl1.SetSaveResults(txtoutputimage.Text);


            axFaceDetectionControl1.SetMinNeighbors(Int32.Parse(cbomingeighors.Text));
            axFaceDetectionControl1.SetScaleFactor(Double.Parse(txtscalefactor.Text));
            axFaceDetectionControl1.SetMinSize(Int32.Parse(txtminsizewidth.Text), Int32.Parse(txtminsizeheight.Text));

            axFaceDetectionControl1.SetMaxSize(Int32.Parse(txtmaxsizewidth.Text), Int32.Parse(txtmaxsizeheight.Text));


            BlackList();
            ControlCountFace();



            System.IO.FileStream fs;
            //Specify a valid picture file path on your computer.

            if (txtoutputimage.Text != "")
            {
                if (System.IO.File.Exists(txtoutputimage.Text))
                {
                    fs = new System.IO.FileStream(txtoutputimage.Text,
                    System.IO.FileMode.Open, System.IO.FileAccess.Read);
                    PictureBox1.Image = System.Drawing.Image.FromStream(fs);
                    fs.Close();
                }
            }



            ListBox1.Items.Clear();
            int ix, iy, iwidth, iheight;



            for (int i = 0; i < axFaceDetectionControl1.DetectedRectGetCount(); i++)
            {
                ix = axFaceDetectionControl1.DetectedRectGetX((short)i);
                iy = axFaceDetectionControl1.DetectedRectGetY((short)i);
                iwidth = axFaceDetectionControl1.DetectedRectGetWidth((short)i);
                iheight = axFaceDetectionControl1.DetectedRectGetHeight((short)i);

                ListBox1.Items.Add(ix.ToString() + " " + iy.ToString() + " " + iwidth.ToString() + " " + iheight.ToString());
            }

            btnblacklist.Enabled = false;


        }
    }
}

File: vb.net2010/Form1.vb
Public Class Form1

    Dim strApp As String
    Private Sub Button1_Click(sender As System.Object, e As System.EventArgs) Handles Button1.Click


        OpenFileDialog1.Filter = "All Files (*.*)|*.*|JPEG (*.jpg)|*.jpg|PNG (*.png)|*.png|Bitmap (*.bmp)|*.bmp|TIF (*.tif)|*.tif"

        If OpenFileDialog1.ShowDialog() = DialogResult.OK Then

            txtsourceimage.Text = OpenFileDialog1.FileName

        End If

    End Sub

    Private Sub ControlCountFace()
        If RadioButton1.Checked Then

            lbltotalface.Text = AxFaceDetectionControl1.CountFaces(strApp + "\demo1.png")

        End If

        If RadioButton2.Checked Then

            lbltotalface.Text = AxFaceDetectionControl1.CountFaces(strApp + "\demo2.png")

        End If

        If RadioButton3.Checked Then

            lbltotalface.Text = AxFaceDetectionControl1.CountFaces(txtsourceimage.Text)

        End If

    End Sub
    Private Sub BlackList()
        Dim ix As Integer
        Dim iy As Integer
        Dim iwidth As Integer
        Dim iheight As Integer

        AxFaceDetectionControl1.ClearBlackList()

        For i = 0 To AxFaceDetectionControl1.DetectedRectGetCount - 1

            ix = AxFaceDetectionControl1.DetectedRectGetX(i)
            iy = AxFaceDetectionControl1.DetectedRectGetY(i)
            iwidth = AxFaceDetectionControl1.DetectedRectGetWidth(i)
            iheight = AxFaceDetectionControl1.DetectedRectGetHeight(i)

            If i = 0 Or i = 1 Then

                AxFaceDetectionControl1.Add2BlackList(ix, iy, iwidth, iheight)

            End If
        Next

    End Sub
    Private Sub Button2_Click(sender As System.Object, e As System.EventArgs) Handles Button2.Click
        If RadioButton3.Checked Then
            If txtsourceimage.Text = "" Then
                MessageBox.Show("Please input source image first")
                Exit Sub

            End If
        End If

        If chkoutputdetectedimage.Checked Then
            If txtoutputimage.Text = "" Then

                MessageBox.Show("Please select the path of output detected image first")
                Exit Sub

            End If
        End If


        AxFaceDetectionControl1.SetFrontalFaceCascade(txtfrontalcascade.Text)
        AxFaceDetectionControl1.SetProfileFaceCascade(txtprofilecascade.Text)

        AxFaceDetectionControl1.EnableEqualizeHistogram = chkequalize.Checked

        AxFaceDetectionControl1.EnableFrontalFace = chkfrontalface.Checked
        AxFaceDetectionControl1.EnableProfileFace = chkprofileface.Checked
        AxFaceDetectionControl1.EnableTiltedFrontalFace = chktiltedfrontalface.Checked
        AxFaceDetectionControl1.EnableTiltedProfileFace = chktiltedprofileface.Checked
       

        If txtoutputimage.Text <> "" Then AxFaceDetectionControl1.SetSaveResults(txtoutputimage.Text)


        AxFaceDetectionControl1.SetMinNeighbors(cbomingeighors.Text)
        AxFaceDetectionControl1.SetScaleFactor(txtscalefactor.Text)
        AxFaceDetectionControl1.SetMinSize(txtminsizewidth.Text, txtminsizeheight.Text)

        AxFaceDetectionControl1.SetMaxSize(txtmaxsizewidth.Text, txtmaxsizeheight.Text)


        AxFaceDetectionControl1.ClearBlackList()
        ControlCountFace()
       


        Dim fs As System.IO.FileStream
        ' Specify a valid picture file path on your computer.

        If txtoutputimage.Text <> "" Then
            If System.IO.File.Exists(txtoutputimage.Text) Then
                fs = New System.IO.FileStream(txtoutputimage.Text,
                IO.FileMode.Open, IO.FileAccess.Read)
                PictureBox1.Image = System.Drawing.Image.FromStream(fs)
                fs.Close()
            End If
        End If



        ListBox1.Items.Clear()
        Dim ix, iy, iwidth, iheight As Integer

        For i = 0 To AxFaceDetectionControl1.DetectedRectGetCount - 1

            ix = AxFaceDetectionControl1.DetectedRectGetX(i)
            iy = AxFaceDetectionControl1.DetectedRectGetY(i)
            iwidth = AxFaceDetectionControl1.DetectedRectGetWidth(i)
            iheight = AxFaceDetectionControl1.DetectedRectGetHeight(i)

            ListBox1.Items.Add(Str(ix) + " " + Str(iy) + " " + Str(iwidth) + " " + Str(iheight))
        Next

        btnblacklist.Enabled = True

    End Sub

    Private Sub Form1_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load

        Dim iFindIndex As Integer
        strApp = Application.ExecutablePath
        iFindIndex = strApp.IndexOf("Examples")

        If iFindIndex <> -1 Then

        strApp = strApp.Substring(0, iFindIndex)

        strApp = strApp + "FacePhoto"


        cbomingeighors.Items.Add("1")
        cbomingeighors.Items.Add("2")
        cbomingeighors.Items.Add("3")
        cbomingeighors.Items.Add("4")
        cbomingeighors.Items.Add("5")
        cbomingeighors.SelectedIndex = 3



        End If




    End Sub

    Private Sub Button3_Click(sender As System.Object, e As System.EventArgs) Handles Button3.Click
        SaveFileDialog1.Filter = "PNG (*.png)|*.png"

        If SaveFileDialog1.ShowDialog() = DialogResult.OK Then

            txtoutputimage.Text = SaveFileDialog1.FileName
        End If
    End Sub

    Private Sub chkoutputdetectedimage_Click(sender As System.Object, e As System.EventArgs) Handles chkoutputdetectedimage.Click
        If chkoutputdetectedimage.Checked Then
            txtoutputimage.Enabled = True
            Button3.Enabled = True
        Else
            txtoutputimage.Enabled = False
            Button3.Enabled = False
        End If
    End Sub

    Private Sub RadioButton3_Click(sender As System.Object, e As System.EventArgs) Handles RadioButton3.Click
        txtmaxsizewidth.Text = 0
        txtmaxsizeheight.Text = 0
        Button1.Enabled = True
    End Sub

    Private Sub RadioButton2_Click(sender As System.Object, e As System.EventArgs) Handles RadioButton2.Click
        txtmaxsizewidth.Text = 50 'Objects larger than 50x50 that are ignored
        txtmaxsizeheight.Text = 50
        Button1.Enabled = False

    End Sub

    Private Sub RadioButton1_Click(sender As System.Object, e As System.EventArgs) Handles RadioButton1.Click
       txtmaxsizewidth.Text = 0
        txtmaxsizeheight.Text = 0
        Button1.Enabled = False

    End Sub

    Private Sub Button4_Click(sender As System.Object, e As System.EventArgs) Handles btnblacklist.Click
        If RadioButton3.Checked Then
            If txtsourceimage.Text = "" Then
                MessageBox.Show("Please input source image first")
                Exit Sub

            End If
        End If

        If chkoutputdetectedimage.Checked Then
            If txtoutputimage.Text = "" Then

                MessageBox.Show("Please select the path of output detected image first")
                Exit Sub

            End If
        End If


        AxFaceDetectionControl1.SetFrontalFaceCascade(txtfrontalcascade.Text)
        AxFaceDetectionControl1.SetProfileFaceCascade(txtprofilecascade.Text)

           AxFaceDetectionControl1.EnableEqualizeHistogram = chkequalize.Checked

        AxFaceDetectionControl1.EnableFrontalFace = chkfrontalface.Checked
        AxFaceDetectionControl1.EnableProfileFace = chkprofileface.Checked
        AxFaceDetectionControl1.EnableTiltedFrontalFace = chktiltedfrontalface.Checked
        AxFaceDetectionControl1.EnableTiltedProfileFace = chktiltedprofileface.Checked

        If txtoutputimage.Text <> "" Then AxFaceDetectionControl1.SetSaveResults(txtoutputimage.Text)


        AxFaceDetectionControl1.SetMinNeighbors(cbomingeighors.Text)
        AxFaceDetectionControl1.SetScaleFactor(txtscalefactor.Text)
        AxFaceDetectionControl1.SetMinSize(txtminsizewidth.Text, txtminsizeheight.Text)

        AxFaceDetectionControl1.SetMaxSize(txtmaxsizewidth.Text, txtmaxsizeheight.Text)


        BlackList()
        ControlCountFace()



        Dim fs As System.IO.FileStream
        ' Specify a valid picture file path on your computer.

        If txtoutputimage.Text <> "" Then
            If System.IO.File.Exists(txtoutputimage.Text) Then
                fs = New System.IO.FileStream(txtoutputimage.Text,
                IO.FileMode.Open, IO.FileAccess.Read)
                PictureBox1.Image = System.Drawing.Image.FromStream(fs)
                fs.Close()
            End If
        End If



        ListBox1.Items.Clear()
        Dim ix, iy, iwidth, iheight As Integer

        For i = 0 To AxFaceDetectionControl1.DetectedRectGetCount - 1

            ix = AxFaceDetectionControl1.DetectedRectGetX(i)
            iy = AxFaceDetectionControl1.DetectedRectGetY(i)
            iwidth = AxFaceDetectionControl1.DetectedRectGetWidth(i)
            iheight = AxFaceDetectionControl1.DetectedRectGetHeight(i)

            ListBox1.Items.Add(Str(ix) + " " + Str(iy) + " " + Str(iwidth) + " " + Str(iheight))
        Next

        btnblacklist.Enabled = False

    End Sub
End Class
